kep/260-network-aware-scheduling/README.md

```diff
@@ -0,0 +1,1398 @@
+type NetworkTopologySpec struct {
+	// The manual defined weights of the cluster
+	Weights WeightList `json:"weights,omitempty"`
+}
+
+// WeightList contains an array of WeightInfo objects.
+// +protobuf=true
+type WeightList []WeightInfo
+
+// CostList contains an array of OriginInfo objects.
+// +protobuf=true
+type CostList []OriginInfo
+
+// WeightInfo contains information about all network costs for a given algorithm.
+// +protobuf=true
+type WeightInfo struct {
+	// The name of the weightInfo (e.g., UserDefined)
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+
+	// Costs between regions
+	RegionCostList CostList `json:"regionCostList" protobuf:"bytes,2,opt,name=regionCostList"`
+
+	// Costs between zones
+	ZoneCostList CostList `json:"zoneCostList" protobuf:"bytes,3,opt,name=zoneCostList"`
+}
+
+// OriginInfo contains information about network costs for a particular Origin.
+// +protobuf=true
+type OriginInfo struct {
+	// The name of the origin (e.g., Region Name, Zone Name).
+	Origin string `json:"origin" protobuf:"bytes,1,opt,name=origin"`
+
+	// Costs for the particular origin.
+	Costs []CostInfo `json:"destination" protobuf:"bytes,2,opt,name=destination"`
+}
+
+// CostInfo contains information about networkCosts.
+// +protobuf=true
+type CostInfo struct {
+	// Name of the destination (e.g., Region Name, Zone Name).
+	Destination string `json:"destination" protobuf:"bytes,1,opt,name=destination"`
+
+	// Bandwidth capacity between origin and destination. 
+	// +optional
+	BandwidthCapacity resource.Quantity `json:"bandwidthCapacity" protobuf:"bytes,2,opt,name=bandwidthCapacity"`
+
+	// Bandwidth allocatable between origin and destination.
+	// +optional
+	BandwidthAllocatable resource.Quantity `json:"bandwidthAllocatable" protobuf:"bytes,3,opt,name=bandwidthAllocatable"`
+
+	// Network Cost between origin and destination
+	NetworkCost int64 `json:"networkCost" protobuf:"bytes,4,opt,name=networkCost"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NetworkTopologyList is a collection of netTopologies.
+type NetworkTopologyList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	// Items is the list of AppGroup
+	Items []NetworkTopology `json:"items"`
+}
+```
+
+## The inclusion of bandwidth in the scheduling process 
+
+### Bandwidth Requests via extended resources
+
+We plan to advertise the node's (physical) bandwidth capacity of the cluster via [extended resources](https://kubernetes.io/docs/tasks/administer-cluster/extended-resource-node/). 
+Bandwidth resources will be advertised as extended resources based on a bandwidth resource component open-sourced [here](https://github.com/jpedro1992/bandwidth-as-an-extended-resource). 
+These component will send HTTP requests to the Kubernetes API server regarding the node's bandwidth capacity. 
```

Comment:
Here, we are considering static since we don't believe the bandwidth capacity of the physical node would change a lot. For that, you would need to add / modify its physical network interface.
So, as mentioned above, the node's bandwidth capacity will be managed via extended resources while between regions/zones, we plan to have a controller that based on allocated pods will update the allocatable bandwidth accordingly.